src/libcommonserver/log/customrollingfileappender.cpp

```diff
@@ -286,30 +288,45 @@ void CustomRollingFileAppender::rollover(bool alreadyLocked) {
 }
 
 void CustomRollingFileAppender::checkForExpiredFiles() {
-    // Delete expired files
-    if (_expire > 0) {
-        _lastExpireCheck = std::chrono::system_clock::now();
-        QDateTime now = QDateTime::currentDateTime();
-
-        IoError ioError = IoErrorSuccess;
-        SyncPath logDirPath;
-        if (!IoHelper::logDirectoryPath(logDirPath, ioError)) {
-            throw std::runtime_error("Error in CustomRollingFileAppender: failed to get the log directory path.");
+    _lastExpireCheck = std::chrono::system_clock::now();
+    // Archive Previous Log Files and delete expired files
+    IoError ioError = IoErrorSuccess;
+    SyncPath logDirPath;
+    if (!IoHelper::logDirectoryPath(logDirPath, ioError) || ioError != IoErrorSuccess) {
+        throw std::runtime_error("Error in CustomRollingFileAppender: failed to get the log directory path.");
```

Comment:
Is this properly catched? If not, do we really the app to crash in such a case?

Reply:
[2f69544](https://github.com/Infomaniak/desktop-kDrive/pull/159/commits/2f69544f6a274ab3994e29ce9a1e9d287e15910d)